src/main.rs

```diff
@@ -73,68 +73,60 @@ impl Process {
     }
 }
 
-fn env_version(env_var_name: &str) -> Result<Option<PackageVersion>> {
-    let version = if let Some(raw_version) = env::var_os(env_var_name) {
-        Some(PackageVersion::new(
-            raw_version
-                .into_string()
-                .map_err(|raw| {
-                    anyhow!("Failed to interpret {env_var_name} {raw:?} as UTF-8 string.")
-                })?
-                .as_str(),
-        )?)
+fn env_version(env_var_name: &str) -> Result<Option<String>> {
+    if let Some(raw_version) = env::var_os(env_var_name) {
+        Ok(Some(raw_version.into_string().map_err(|raw| {
+            anyhow!("Failed to interpret {env_var_name} {raw:?} as UTF-8 string.")
+        })?))
     } else {
-        None
-    };
-    Ok(version)
+        Ok(None)
+    }
+}
+
+fn find_pants_installation() -> Result<Option<PantsConfig>> {
+    if let Ok(build_root) = BuildRoot::find(None) {
+        if let Some(pants_bootstrap) = PantsBootstrap::load(&build_root)? {
+            pants_bootstrap.export_env();
+        }
+        let pants_config = PantsConfig::parse(build_root)?;
+        return Ok(Some(pants_config));
+    }
+    Ok(None)
 }
 
 #[time("debug", "scie-pants::{}")]
 fn get_pants_process() -> Result<Process> {
-    let build_root = BuildRoot::find(None)?;
-    if let Some(pants_bootstrap) = PantsBootstrap::load(&build_root)? {
-        pants_bootstrap.export_env();
-    }
-    let pants_config = PantsConfig::parse(build_root)?;
-    let build_root = pants_config.build_root().to_path_buf();
+    let pants_installation = find_pants_installation()?;
+    let (build_root, configured_pants_version, debugpy_version) =
+        if let Some(ref pants_config) = pants_installation {
+            (
+                Some(pants_config.build_root().to_path_buf()),
+                pants_config.package_version(),
+                pants_config.debugpy_version(),
+            )
+        } else {
+            (None, None, None)
+        };
 
     let env_pants_sha = env_version("PANTS_SHA")?;
     let env_pants_version = env_version("PANTS_VERSION")?;
     if let (Some(pants_sha), Some(pants_version)) = (&env_pants_sha, &env_pants_version) {
         bail!(
             "Both PANTS_SHA={pants_sha} and PANTS_VERSION={pants_version} were set. \
             Please choose one.",
-            pants_sha = pants_sha.original,
-            pants_version = pants_version.original
         )
     }
 
-    let pants_version = if let Some(ref env_version) = env_pants_version {
+    let pants_version = if let Some(env_version) = env_pants_version {
         Some(env_version)
     } else if env_pants_sha.is_none() {
-        pants_config.package_version()
+        configured_pants_version
     } else {
         None
     };
 
-    let debugpy_version = pants_config.debugpy_version().unwrap_or_default();
-
-    let python = if let Some(version) = pants_version {
```

Comment:
This logic moved into the `configure` binding. Any move like this is good - the less the scie-pants binary does, the better. It's in the hot path and ideally should know as little about Pants as possible.